filebeat/module/postgresql/csv/_meta/fields.yml

```diff
@@ -0,0 +1,71 @@
+- name: log
+  type: group
+  description: >
+    Fields from the PostgreSQL CSV log files.
+  fields:
+    - name: database
+      example: "postgres"
+      description: Name of database.
+    - name: client_addr
+      example: "127.0.0.1"
+      description: Host where the connection originated from.
+    - name: client_port
+      example: "59700"
+      description: Port where the connection originated from.
+    - name: session_id
+      description: PostgreSQL session.
+      example: "5ff1dd98.22"
+    - name: session_line_num
+      type: long
+      description: >
+        Line number inside a session.
+        (session_id,session_line_num) identifies uniquely an event.
+    - name: command_tag
+      example: "SELECT"
+      description: >
+        Type of session's current command.
+        The complete list can be found at: src/include/tcop/cmdtaglist.h
+    - name: session_start_time
+      type: date
+      description: Time when this session started.
+    - name: virtual_transaction_id
+      description: Backend local transaction id .
+    - name: transaction_id
+      type: long
+      description: The id of current transaction.
+    - name: error_severity
+      example: "LOG"
+      description: Valid values are DEBUG5, DEBUG4, DEBUG3, DEBUG2, DEBUG1, INFO, NOTICE, WARNING, ERROR, LOG, FATAL, and PANIC. 
+    - name: sql_state_code
+      description: >
+        State code returned by Postgres (if any).
+        This code is not a number.
+        See also https://www.postgresql.org/docs/current/errcodes-appendix.html
+    - name: message
+      description: Main message of this event.
+    - name: detail
+    - name: hint
+    - name: internal_query
+      description: internal query that led to the error (if any).
+    - name: internal_query_pos
+      type: long
+      description: character count of the error position (if any).
+    - name: context
+      description: Error context.
+    - name: query
+      description: User query that led to the error (if any and enabled by log_min_error_statement).
+    - name: query_pos
+      type: long
+      description: character count of the error position (if any).
+    - name: location
+      description: Location of the error in the PostgreSQL source code (if log_error_verbosity is set to verbose).
+    - name: application_name
+      description: Name of the application of this event. It is defined by the client.
+    - name: backend_type
+      example: "client backend"
+      description: > 
+        Type of backend of this event.
+        Possible types are autovacuum launcher, autovacuum worker, logical replication launcher,
+        logical replication worker, parallel worker, background writer, client backend, checkpointer,
+        startup, walreceiver, walsender and walwriter.
+        In addition, background workers registered by extensions may have additional types.
```

Comment:
There is no need to add these fields, they are already documented in the `log` metricset. We could consider moving their definition to the module-level `fields.yml`, but by now, let's just avoid duplicating them.

I don't know though if we have another fileset that collects the same fields as other fileset.


```
```

Reply:
I'm in doubt here. Not all fields are defined in the postgresql.log fileset.

So, should I move duplicates to beats/filebeat/module/postgresql/_meta/fields.yml` and leave others inside the fileset? Is there any example around the code?

Reply:
Ok, I found an example in the Kafka module in Metricbeat, there we have some partition fields that are filled by multiple metricsets, they are defined at the module level, here:
https://github.com/elastic/beats/blob/8e8bfc8786abded0bca05da7e7cbb8751b96061b/metricbeat/module/kafka/_meta/fields.yml#L38-L51

With filebeat modules it should work the same, we can put common fields at the module level.

Reply:
Since I merged `csv` inside `log`, this part is solved.